add_node.rb

```diff
@@ -25,24 +55,43 @@
 	puts "Enter department: "
 	department = gets.chomp
 
-	puts "Enter semester (mid spring 2016): "
-	semester = gets.chomp
+	if batch_semester.empty?
```

Comment:
Hey, this variable will never be empty as it was initialised to a non-empty value. Check the conditional flow up-top and change the initialisation.